addons/mail/static/src/models/chatter_topbar.js

```diff
@@ -1,12 +1,42 @@
 /** @odoo-module **/
 
 import { registerModel } from '@mail/model/model_core';
-import { one } from '@mail/model/model_field';
+import { attr, one } from '@mail/model/model_field';
+
+import { sprintf } from '@web/core/utils/strings';
 
 registerModel({
     name: 'ChatterTopbar',
     identifyingFields: ['chatter'],
+    recordMethods: {
+        /**
+         * @private
+         * @returns {string}
+         */
+        _computeAttachmentButtonText() {
+            if (!this.chatter || !this.chatter.thread) {
+                return;
+            }
+            const attachments = this.chatter.thread.allAttachments;
+            if (attachments.length === 0) {
+                return this.env._t("Attach files");
+            }
+            return sprintf(
+                this.env._t("%(attachmentCount)s %(attachmentText)s"),
+                {
+                    attachmentCount: attachments.length,
+                    attachmentText: this.env._t("files"),
+                }
+            );
```

Comment:
There is still a problem here with the translation / values.
It currently shows:
`1 files` -> it should be "1 file" instead

We need to be able to handle plural forms differently.

Some languages have a different word(s) for 1 value, 2 value, and many values.

So we need 3 different versions of the translation string:
```
if 1:
  return _t("1 file")
if 2:
  return _t("2 files")
return _t("%(attachmentCount) files")

```
I hope this helps understand. A bit like `_getTitle` as an example in voip DialingPanel.

Reply:
Done!!